File: semestru 4/MotoProjectCs/services/IService.cs
using moto.model;

namespace moto.services;

public interface IService {
    public User FindByUsername(String username);
    public void Login(User user,IObserver client);
    public void Logout(User user,IObserver client);
    public void Add(Player p,int raceId,IObserver client);
    public Team? GetTeamByName(String name);
    public Race[] GetAllRaces();
    public Player[] GetRacePlayersOfTeam(int raceId, int teamId);

}

File: semestru 4/MotoGRPC/Service/src/main/java/Services/IService.java
//package Services;
//
//
//import model.Player;
//import model.Race;
//import model.Team;
//import model.User;
//
//import java.util.List;
//import java.util.Map;
//import java.util.Optional;
//
//public interface IService {
//    public void login(User user,IObserver client) throws MotoException;
//    public void logout(User user,IObserver client) throws MotoException;
//    public void add(Player p,Integer raceId,IObserver client) throws MotoException;
//    public Optional<Team> getTeamByName(String name) throws MotoException;
//    public Race[] getAllRaces() throws MotoException;
//    public Player[] getRacePlayersOfTeam(Integer raceId, Integer teamId) throws MotoException;
//
//}
